iree/compiler/Dialect/HAL/Target/LLVM/internal/EmbeddedLinkerTool.cpp

```diff
@@ -88,8 +93,11 @@ class EmbeddedLinkerTool : public LinkerTool {
     }
     artifacts.libraryFile.close();
 
+    std::string embeddedToolPath = getEmbeddedToolPath();
+    if (embeddedToolPath.empty()) return llvm::None;
```

Comment:
Together with the other changes, this early exit changes the error message from something like

```
LLD (ld.lld) not found on path; specify with the IREE_LLVMAOT_EMBEDDED_LINKER_PATH environment variable or -iree-llvm-embedded-linker-path=
sh: line 1: -flavor: command not found
Linking failed; escaped command line returned exit code 32512:

 -flavor gnu -o /tmp/simple_mul_dispatch_0-ed0c39.so -nostdlib -static -shared --gc-sections -z now -z relro --discard-all --hash-style=sysv /tmp/simple_mul_dispatch_0-ed0c39.o /tmp/simple_mul_dispatch_0-ed0c39.librt.o /tmp/simple_mul_dispatch_0-ed0c39.librt64.o

/usr/local/google/home/kooljblack/Code/iree-data/simple_mul.mlir:3:8: error: failed to link executable and generate target dylib using linker toolchain 
  %0 = "arith.mulf"(%arg0, %arg1) {name = "mul.1"} : (tensor<4xf32>, tensor<4xf32>) -> tensor<4xf32>
       ^
```

to

```
error: required embedded linker tool (typically `lld`) not found after searching:
  * -iree-llvm-embedded-linker-path= flag
  * IREE_LLVMAOT_EMBEDDED_LINKER_PATH environment variable
  * common locations at relative file paths
  * system PATH
Run with -debug-only=llvmaot-linker for search details
D:\dev\projects\iree/iree/samples/models/simple_abs.mlir:2:13: error: failed to link executable and generate target dylib (check above for more specific error messages)
  %result = math.abs %input : tensor<f32>
            ^
```

(` -flavor gnu -o ...` doesn't mean much without knowing that there should be a `lld` executable at the start of that command :P)

Reply:
Note that this still checks for the linker tool after other parts of compilation have ran, and it still outputs the full "`note: see current operation: "hal.executable.variant"()` {giant MLIR blob here}", so users/developers will still have to scroll up in the error output to see this :/